symengine/functions.cpp

```diff
@@ -42,6 +42,64 @@ extern umap_basic_basic inverse_cst;
 
 extern umap_basic_basic inverse_tct;
 
+Conjugate::Conjugate(const RCP<const Basic> &arg) : OneArgFunction(arg)
+{
+    SYMENGINE_ASSIGN_TYPEID()
+    SYMENGINE_ASSERT(is_canonical(arg))
+}
+
+bool Conjugate::is_canonical(const RCP<const Basic> &arg) const
+{
+    if (is_a_Number(*arg)) {
+        if (eq(*arg, *ComplexInf)) {
+            return true;
+        }
+        return false;
+    }
+    if (is_a<Constant>(*arg)) {
+        return false;
+    }
+    if (is_a<Mul>(*arg)) {
+        return false;
+    }
+    return true;
+}
+
+RCP<const Basic> Conjugate::create(const RCP<const Basic> &arg) const
+{
+    return conjugate(arg);
+}
+
+RCP<const Basic> conjugate(const RCP<const Basic> &arg)
+{
+    if (is_a_Number(*arg)) {
+        if (not down_cast<const Number &>(*arg).is_complex()) {
+            return arg;
+        }
+        if (is_a_Complex(*arg)) {
+            return down_cast<const ComplexBase &>(*arg).conjugate();
+        }
+    }
+    if (is_a<Constant>(*arg)) {
+        return arg;
+    }
+    if (is_a<Mul>(*arg)) {
+        RCP<const Basic> coef
+            = conjugate(down_cast<const Mul &>(*arg).get_coef());
+        map_basic_basic dict = down_cast<const Mul &>(*arg).get_dict();
+        RCP<const Basic> s = one;
+        for (const auto &p : dict) {
+            if (is_a<Integer>(*p.second)) {
+                s = mul(s, pow(conjugate(p.first), p.second));
+            } else {
+                s = mul(s, conjugate(pow(p.first, p.second)));
+            }
+        }
+        return mul(coef, s);
```

Comment:
Don't use `mul` repeatedly. You can use the helper functions in Mul to create a Mul directly.